### ISSUE TITLE
PoX: Stacker reward does not depend on amount stacked

### ISSUE THREAD

**Issue by psq** [blockstack/stacks-blockchain#1924]
I registered 2 addresses and registered them with `stack-stx`, one with 2_000_000_000_000, for 5 cycles, and one with 6_100_000_000_000, expecting the second address would get 3 times the reward as the minimum for stacking was about 2_000_000_000_000 and a bit (I forgot to account for the increasing supply with new STX minted each block on the first address, and was not expecting to get rewards, but did anyway (maybe that's the 25% case kicking in, as no one else was stacking).

Here's the data from calling the pox contract (`get-reward-set-size`, `get-reward-set-pox-address`), both addresses are mine, and no one else was stacking, so that makes it easier to track.

```
Cycle 0: 0 0-119 active
Cycle 1: 0 120-239 active
Cycle 2: 0 240-359 active
Cycle 3: 0 360-479 active
Cycle 4: 0 480-599 active
Cycle 5: 1 600-719 active
==> 5 0 ST1TWA18TSWGDAFZT377THRQQ451D1MSEM69C761 mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH 2000000000000
Cycle 6: 1 720-839 active
==> 6 0 ST1TWA18TSWGDAFZT377THRQQ451D1MSEM69C761 mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH 2000000000000
Cycle 7: 2 840-959 active
==> 7 0 ST1TWA18TSWGDAFZT377THRQQ451D1MSEM69C761 mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH 2000000000000
==> 7 1 ST30X92ZMEVF8W5FCPVX3CXFCH230496K229ZCC4X myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG 6100000000000
Cycle 8: 2 960-1079 active
==> 8 0 ST1TWA18TSWGDAFZT377THRQQ451D1MSEM69C761 mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH 2000000000000
==> 8 1 ST30X92ZMEVF8W5FCPVX3CXFCH230496K229ZCC4X myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG 6100000000000
Cycle 9: 2 1080-1199 active
==> 9 0 ST1TWA18TSWGDAFZT377THRQQ451D1MSEM69C761 mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH 2000000000000
==> 9 1 ST30X92ZMEVF8W5FCPVX3CXFCH230496K229ZCC4X myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG 6100000000000
Cycle 10: 1 1200-1319 active
==> 10 0 ST30X92ZMEVF8W5FCPVX3CXFCH230496K229ZCC4X myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG 6100000000000
Cycle 11: 1 1320-1439 active
==> 11 0 ST30X92ZMEVF8W5FCPVX3CXFCH230496K229ZCC4X myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG 6100000000000
Cycle 12: 0 1440-1559 active
Cycle 13: 0 1560-1679 active
Cycle 14: 0 1680-1799 active
Cycle 15: 0 1800-1919 active
Cycle 16: 0 1920-2039 active
Cycle 17: 0 2040-2159 active
Cycle 18: 0 2160-2279 active
Cycle 19: 0 2280-2399 active
```

And here are the rewards (from `listunspent`)
```javascript
{
"mfrQxG6v9DVhUVmSQiV9Ac7NcQiXS4BUGH": [
  { confirmations: 794, amount: 0.0001 },
  { confirmations: 794, amount: 0.0001 },
  { confirmations: 794, amount: 0.0001 },
  { confirmations: 794, amount: 0.0001 },
  { confirmations: 794, amount: 0.0001 },
  { confirmations: 674, amount: 0.0001 },
  { confirmations: 674, amount: 0.0001 },
  { confirmations: 674, amount: 0.0001 },
  { confirmations: 674, amount: 0.0001 },
  { confirmations: 674, amount: 0.0001 },
  { confirmations: 554, amount: 0.0001 },
  { confirmations: 554, amount: 0.0001 },
  { confirmations: 554, amount: 0.0001 },
  { confirmations: 554, amount: 0.0001 },
  { confirmations: 433, amount: 0.0001 },
  { confirmations: 433, amount: 0.0001 },
  { confirmations: 433, amount: 0.0001 },
  { confirmations: 433, amount: 0.0001 },
  { confirmations: 314, amount: 0.0001 },
  { confirmations: 314, amount: 0.0001 },
  { confirmations: 314, amount: 0.0001 },
  { confirmations: 314, amount: 0.0001 },
  { confirmations: 314, amount: 0.0001 }
]
"myBqL5eC3tSZ7HBUYAmi8773NXAuST3PBG": [
  { confirmations: 553, amount: 0.0001 },
  { confirmations: 553, amount: 0.0001 },
  { confirmations: 553, amount: 0.0001 },
  { confirmations: 553, amount: 0.0001 },
  { confirmations: 434, amount: 0.0001 },
  { confirmations: 434, amount: 0.0001 },
  { confirmations: 434, amount: 0.0001 },
  { confirmations: 434, amount: 0.0001 },
  { confirmations: 313, amount: 0.0001 },
  { confirmations: 313, amount: 0.0001 },
  { confirmations: 313, amount: 0.0001 },
  { confirmations: 313, amount: 0.0001 },
  { confirmations: 313, amount: 0.0001 },
  { confirmations: 194, amount: 0.0001 },
  { confirmations: 194, amount: 0.0001 },
  { confirmations: 194, amount: 0.0001 },
  { confirmations: 194, amount: 0.0001 },
  { confirmations: 194, amount: 0.0001 },
  { confirmations: 74, amount: 0.0001 },
  { confirmations: 74, amount: 0.0001 },
  { confirmations: 74, amount: 0.0001 },
  { confirmations: 74, amount: 0.0001 },
  { confirmations: 74, amount: 0.0001 }
]
}
```
you can see that when the 2 addresses stacking overlaps, they get the same number of rewards (1 per cycle, roughly 120 blocks apart), confirmation 553/554, 433/434, 313/314.

cc @jcnelson

**Comment by diwakergupta** [comment]
@kantai this should be easy to verify via an integration test?

**Comment by kantai** [comment]
This is definitely the case in the current testnet implementation, and is one of the items I'll be addressing in the ongoing PoX work (this is ultimately related to the threshold adjustment).

**Comment by kantai** [comment]
This is closed by #1954, and has been deployed to Krypton.
